File: 算法训练营/03-数组链表跳表/189/暴力迭代.go
/*
 * @lc app=leetcode.cn id=189 lang=golang
 *
 * [189] 旋转数组
 */

// @lc code=start
func rotate(nums []int, k int)  {
	length := len(nums)
	for i := 0; i<k; i++ {
		tmp := nums[0]
		for j := 0; j<length; j++ {
			next := j+1
			if next == length {
				next %= length
			}
			tmp, nums[next] = nums[next], tmp
		}
	}
}

// @lc code=end

File: LeetCode/0496/暴力迭代.py
#
# @lc app=leetcode.cn id=496 lang=python3
#
# [496] 下一个更大元素 I
#

# @lc code=start
class Solution:
    def nextGreaterElement(self, nums1: List[int], nums2: List[int]) -> List[int]:
        res = []
        for n in nums1:
            maxv, start = -1, False
            for m in nums2:
                if n == m:
                    start = True
                
                if start and m > n:
                    maxv = m   
                    break
            res.append(maxv)
        return res
                
# @lc code=end